### PR TITLE
TIG-3112 Prompt user to add ssh key with passphrase to ssh-agent to run the script

### PR DESCRIPTION
**Pull request by MikhailShchatko**

### CODE REVIEW

File: virtual_workstation_setup.sh [hunk 1]
```diff
@@ -12,6 +12,27 @@ silent_grep() {
     command grep -q  > /dev/null 2>&1 "$@"
 }
 
+set +e
+for filename in ~/.ssh/id_*; do
+    [[ "$filename" == *".pub" ]] && continue
+    # check if SSH key has passphrase
+    ssh-keygen -y -P "" -f "$filename" > /dev/null 2>&1
```
Comment (MikhailShchatko) [inline virtual_workstation_setup.sh:19]: This is to support a new `openssh` format of the keys: https://security.stackexchange.com/a/129729

Comment (guoyr) [reply virtual_workstation_setup.sh:19]: Nice. Also, nice use of `ssh-keygen`

File: virtual_workstation_setup.sh [hunk 2]
```diff
@@ -12,6 +12,27 @@ silent_grep() {
     command grep -q  > /dev/null 2>&1 "$@"
 }
 
+set +e
+for filename in ~/.ssh/id_*; do
```
Comment (guoyr) [inline virtual_workstation_setup.sh:16]: Let's preemptively add a comment saying something like: If the user has a different name for the ssh key, we assume they know about the agent and can set it up manually. This check is for the majority of users.

File: virtual_workstation_setup.sh [hunk 3]
```diff
@@ -12,6 +12,27 @@ silent_grep() {
...
+    [ $? -eq 0 ] && continue
+    # check if ssh agent is running and have any SSH keys added
+    # (we assume that the key with passphrase will be added, but not checking it explicitly)
+    ssh-add -l > /dev/null 2>&1
+    if [ $? -ne 0 ]; then
+        echo "SSH key with passphare found: $filename"
+        echo "Please do the following steps:"
+        echo " - Setup ssh-agent"
+        echo " - Add SSH key with passphrase to the ssh-agent"
+        echo " - Re-run the command"
```
Comment (guoyr) [inline virtual_workstation_setup.sh:29]: 💅  "Rerun the Evergreen project setup command"

File: virtual_workstation_setup.sh [hunk 4]
```diff
@@ -12,6 +12,27 @@ silent_grep() {
...
 }
 
+set +e
+for filename in ~/.ssh/id_*; do
+    [[ "$filename" == *".pub" ]] && continue
+    # check if SSH key has passphrase
+    ssh-keygen -y -P "" -f "$filename" > /dev/null 2>&1
+    [ $? -eq 0 ] && continue
+    # check if ssh agent is running and have any SSH keys added
+    # (we assume that the key with passphrase will be added, but not checking it explicitly)
```
Comment (guoyr) [inline virtual_workstation_setup.sh:22]: 👍

**Review by guoyr — APPROVED** [review]
A few defensive suggestions. Otherwise LGTM!

File: virtual_workstation_setup.sh (hunk 2 above)
Comment (MikhailShchatko) [reply virtual_workstation_setup.sh:18]: done

File: virtual_workstation_setup.sh (hunk 3 above)
Comment (MikhailShchatko) [reply virtual_workstation_setup.sh]: done

**Pull request merged**